src/ess/imaging/io.py

```diff
@@ -113,27 +113,16 @@ def load_nexus_histogram_mode_detector(
 
 
 def _assign_coord_or_make_one(
-    da: sc.DataArray, coord_name: str, coord: sc.Variable | None
+    da: sc.DataArray, dim: str, coord: sc.Variable | None
 ) -> None:
-    da.coords[coord_name] = coord or sc.arange(
-        dim=coord_name, start=0, stop=da.sizes[coord_name]
-    )
-
-
-def _crop_da_by_coord(
-    da: sc.DataArray,
-    dim: str,
-    min_coord: sc.Variable | None,
-    max_coord: sc.Variable | None,
-) -> sc.DataArray:
-    if min_coord is not None and max_coord is not None:
-        return da[dim, min_coord:max_coord]
-    elif min_coord is not None and max_coord is None:
-        return da[dim, min_coord:]
-    elif min_coord is None and max_coord is not None:
-        return da[dim, :max_coord]
+    if (dim in da.coords.keys()) and (coord is not None):
+        da.coords[dim] = coord
+    elif (dim not in da.coords.keys()) and (coord is None):
+        da.coords[dim] = sc.arange(dim=dim, start=0, stop=da.sizes[dim])
```

Comment:
This changed the logic. Now it raises if `(dim not in da.coords) and (coord is not None)`. Previously, it would insert the new coord.

Reply:
Yes, we don't want a new coordinate if it already exists.
I just wanted to make sure it has a coordinate before slicing.
So, if the loaded data already has a coordinate, you can change the cropping range according to that.

Reply:
But that contradicts the code. If a coord exists and `coord is not None`, you overwrite the existing coord. The case I pointed out above would insert a coord if there is non already.

Reply:
Yeah I added that when we want to insert the coordinate but it should be in the loader function not in here.. I'll remove that part....

Reply:
Done...!